File: php/features.php
<?php
declare(strict_types=1);

// MixpanelLexiconSchemas SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/DebugFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class MixpanelLexiconSchemasFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MixpanelLexiconSchemasBaseFeature();
            case "debug":
                return new MixpanelLexiconSchemasDebugFeature();
            case "idempotency":
                return new MixpanelLexiconSchemasIdempotencyFeature();
            case "metrics":
                return new MixpanelLexiconSchemasMetricsFeature();
            case "paging":
                return new MixpanelLexiconSchemasPagingFeature();
            case "ratelimit":
                return new MixpanelLexiconSchemasRatelimitFeature();
            case "retry":
                return new MixpanelLexiconSchemasRetryFeature();
            case "test":
                return new MixpanelLexiconSchemasTestFeature();
            case "timeout":
                return new MixpanelLexiconSchemasTimeoutFeature();
            default:
                return new MixpanelLexiconSchemasBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "debug":
            case "idempotency":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/mixpanellexiconschemas_sdk/features.py
# MixpanelLexiconSchemas SDK feature factory

from mixpanellexiconschemas_sdk.feature.base_feature import MixpanelLexiconSchemasBaseFeature
from mixpanellexiconschemas_sdk.feature.debug_feature import MixpanelLexiconSchemasDebugFeature
from mixpanellexiconschemas_sdk.feature.idempotency_feature import MixpanelLexiconSchemasIdempotencyFeature
from mixpanellexiconschemas_sdk.feature.metrics_feature import MixpanelLexiconSchemasMetricsFeature
from mixpanellexiconschemas_sdk.feature.paging_feature import MixpanelLexiconSchemasPagingFeature
from mixpanellexiconschemas_sdk.feature.ratelimit_feature import MixpanelLexiconSchemasRatelimitFeature
from mixpanellexiconschemas_sdk.feature.retry_feature import MixpanelLexiconSchemasRetryFeature
from mixpanellexiconschemas_sdk.feature.test_feature import MixpanelLexiconSchemasTestFeature
from mixpanellexiconschemas_sdk.feature.timeout_feature import MixpanelLexiconSchemasTimeoutFeature


_FEATURES = {
    "base": lambda: MixpanelLexiconSchemasBaseFeature(),
    "debug": lambda: MixpanelLexiconSchemasDebugFeature(),
    "idempotency": lambda: MixpanelLexiconSchemasIdempotencyFeature(),
    "metrics": lambda: MixpanelLexiconSchemasMetricsFeature(),
    "paging": lambda: MixpanelLexiconSchemasPagingFeature(),
    "ratelimit": lambda: MixpanelLexiconSchemasRatelimitFeature(),
    "retry": lambda: MixpanelLexiconSchemasRetryFeature(),
    "test": lambda: MixpanelLexiconSchemasTestFeature(),
    "timeout": lambda: MixpanelLexiconSchemasTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
